### PR TITLE
Launchpad: Change hover & focus styles on all active links

### LINKED ISSUE #68035: LaunchPad - update "Go to admin" hover color
The "Go to Admin" link should be our standard blue color on hover.

Screen Shot 2022-09-20 at 10 32 18 AM

### PR DESCRIPTION
**Pull request by sharpirate**
#### Proposed changes
This PR makes all active Launchpad links (including "Go to Admin") blue (`--studio-blue-50`) both when hovering and when navigating with the keyboard. Disabled links stay gray.

#### Before

https://user-images.githubusercontent.com/20927667/191381904-f10b7a94-92aa-4570-b542-f721c4a6caaa.mov

#### After

https://user-images.githubusercontent.com/20927667/191381899-9d78c33f-3039-41d6-8c89-37ee581f9127.mov

#### Testing Instructions
1. Go to `http://calypso.localhost:3000/setup/launchpad?flow=newsletter&complete-setup=true&siteSlug=YOUR_SITE_SLUG`
2. Ensure hovering with the mouse and focusing with the keyboard on all active links should turn them blue (`--studio-blue-50`)

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Fixes #68035

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/style.scss [hunk 1]
```diff
@@ -119,8 +119,12 @@
 	padding: 80px 20px 10px;
 	display: none;
 
-	button {
+	.button.navigation-link {
```
Comment (sharpirate) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/style.scss:128]: I've changed the selector because it was insufficient when it comes to CSS specificity

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/style.scss [hunk 2]
```diff
@@ -243,6 +242,11 @@
 	.launchpad__checklist-item-checkmark {
 		fill: var(--studio-blue-50);
 	}
+
+	// change badge color on hover (Choose a Plan step)
+	.badge {
+		background: var(--studio-blue-10);
+	}
```
Comment (Addison-Stavlo) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/style.scss:265]: Noting this css specificity may not be ideal for this style. It looks like its original placement wasn't great either. But it has good behavior in the current state.

Before - it looks like this badge color was set up to change color on any task when hovered. Thats not ideal because when we have a task with a badge that is complete / inert this will still change color on hover. Currently we don't have any like that, but in the future this style will bleed to extra places where unexpected.

Now in this PR - we are making this specific to tasks with `keep-active`. I think this will neglect to take into consideration tasks that are active, incomplete, and not meant to keep-active after completion? - so this style may not take effect on some places where expected in the future.

**Review by Addison-Stavlo — COMMENTED** [review]
This works well!

Just some nitpicking on style rules below that could effect other badge tasks added in the future.

**Comment by edanzer** [comment]
This tests well in Chrome. I don't seem to be able to tab select/focus the main launchpad action links in Firefox or Safari. Wondering if others can check that? Not sure if I should be using different keyboard actions, or if there accessibility attributes we need to add for cross browser support. 

This issue isn't technically about resolving cross-browser accessibility, but if accessible tabbing/focusing is indeed an issue and if the fix is simple, may be worth adding here.

**Comment by Addison-Stavlo** [comment]
Good catch @edanzer !  I am seeing the same thing where we cannot keyboard navigate checklist items outside of Chrome.  We should handle this in a separate PR and I opened an issue here https://github.com/Automattic/wp-calypso/issues/68206

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx [hunk 3]
```diff
@@ -15,7 +15,8 @@ const ChecklistItem = ( { task }: { task: Task } ) => {
 		<li
 			className={ classnames( 'launchpad__task', {
 				'keep-active': keepActive,
-				'is-completed': isCompleted,
+				'is-completed': isCompleted && ! keepActive,
+				pending: ! isCompleted,
```
Comment (sharpirate) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx]: I created a `pending` classname for the tasks that haven't been completed yet. This way we will have a corresponding classname for each of the three statuses: completed, pending and completed tasks that are kept active (tasks you can revisit). This allows for much easier styling since now each task type will have only one classname as opposed to before where we had a task both `is-complete` and `keep-active` which would cause style collisions

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/style.scss (hunk 2 above)
Comment (sharpirate) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/style.scss:265]: @Addison-Stavlo Great catch! I've updated the PR to include the hover and focus badge styles for pending tasks. I've set up a fake badge on a pending task for the purpose of testing:

https://user-images.githubusercontent.com/20927667/192341754-7d131248-13e6-47b2-9f62-80ba0d506151.mov

**Comment by jeyip** [comment]
Unrelated to this PR, but links in the web preview are highlighted when using keyboard navigation. It's misleading because users can't interact with those anchor tags.

**Comment by jeyip** [comment]
## Testing
### Requirements
- [x] Hovering with the mouse and focusing with the keyboard on all active links turns them blue

### Browsers
- [x] Edge
- [x] Firefox
- [x] Chrome

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx (hunk 3 above)
Comment (jeyip) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx]: > This allows for much easier styling since now each task type will have only one classname as opposed to before where we had a task both is-complete and keep-active which would cause style collisions

What happens when `keepActive` is `true` and `isCompleted` is `false`? It looks like we'll have both the `keep-active` and `pending` classnames applied at the same time, which seems to be at odds with the purpose of these changes?

Comment (jeyip) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx]: I feel like these conditions could benefit from some inline documentation, explaining when tasks could be in various states and why. Curious about your thoughts though.

Comment (sharpirate) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx]: @jeyip Great catch about the extra check, I tried to shorten the expression but it's definitely needed. I like the idea of inline documentation as well, I'm starting on these

Comment (edanzer) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx]: I will test just to confirm this works shortly, but also wanted to add a general note. 

I've mentioned in a few other places that it seems like we have a lot of different, related states for completion/active/disabled/etc all interacting, and that the number of states and the different interactions can be quite hard to follow.

Much of that existed before this PR. But the code in question here seems to illustrate it, and I'm wondering if there is a way to simplify this?

Example questions:

```
'keep-active': keepActive && isCompleted, // a task that is completed and can be revisited
'is-completed': isCompleted && ! keepActive, // a task that is completed and can't be revisited
pending: ! isCompleted && ! keepActive, // a task that hasn't been completed yet
```

- We have a keepActive state and keep-active css class. But these don't mean the same thing, because the keep-active css class also requires that isCompleted is true. I would think keep-active would means keep a task active regardless of completion state?
- We have an isCompleted state and is-completed css class, but these don't mean the same thing, because the isCompleted css class also requires keepActive to be false. I would think isCompleted would mean a task is completed, regardless of keepActive state. 
- We're now adding pending as a css class, which is a bit like adding yet another potential state. It says 'a task that has not been completed yet', but would have thought that was basically the default for any task? 
- We also still have taskDisabled, which is not used here in the css classes, but seems to overlap with these states in unclear ways. 

I'm not entirely sure what the solution is here. I think at root, we really still need to simplify the various states for each task. But one though for now for now would this: 
 - simply apply keep-active when keepActive === true
 - simply apply is-complete when isComplete === true
 - handle styling for combinations via css rules based on those two classes (technically, we should be able to style any combination of those two classes based on whether one exists, the other exists, both exist, or neither exist (ie, .keep-active.is-complete would handle situations when both are true).
 - this solution doesn't resolve all the underlying complexity, but it does (maybe) avoid a situation where keep-active and keepActive now mean slightly different things, or where is-complete and isCompleted mean slightly different things.

Comment (sharpirate) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx]: Thanks for taking a look at this @edanzer. I agree that at this point the statuses are created with styling in mind but I think that's fine because that's their only purpose. The reason I decided to add the `pending` classname was because there wasn't an easy way to add a style that is exclusive to only the tasks that are still not completed. We had to target the `.launchpad__task` but that would spill over both the `.is-completed` and `.keep-active` tasks. Then we would have to override the unwanted styles in both of these classes. It's the same when you want certain styles for the `.is-completed` tasks but not the `.keep-active` tasks. Since by default a `.keep-active` task is a `.is-completed` task as well. The benefit of going with the proposed approach is that we can have clean exclusive styles for each task status that won't spill over to the other tasks by default. I believe in the end it's a tradeoff between better semantics and hard styling or worse semantics and better styling. Curious to see your thoughts on this as well @Addison-Stavlo @jeyip

**Comment by edanzer** [comment]
This still tests correctly for me! I have questions raised in other comments about the complexity of our active/completed states, but given those concerns pre-date this PR, I would not consider those concerns blockers. So approving now, and we can decide seprately if we want to address the complexity of state/css classes.

**Review by edanzer — APPROVED** [review]

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx [hunk 4]
```diff
@@ -14,8 +14,9 @@ const ChecklistItem = ( { task }: { task: Task } ) => {
 	return (
 		<li
 			className={ classnames( 'launchpad__task', {
-				'keep-active': keepActive,
-				'is-completed': isCompleted,
+				'keep-active': keepActive && isCompleted, // a task that is completed and can be revisited
```
Comment (jeyip) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx]: Can we switch these conditions around for the sake of readability since the other checks are for `isCompleted` _and then_ `keepActive`?

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx (hunk 3 above)
Comment (jeyip) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx]: > We have a keepActive state and keep-active css class. But these don't mean the same thing

> We have an isCompleted state and is-completed css class, but these don't mean the same thing

> Since by default a .keep-active task is a .is-completed task as well. The benefit of going with the proposed approach is that we can have clean exclusive styles for each task status that won't spill over to the other tasks by default. I believe in the end it's a tradeoff between better semantics and hard styling or worse semantics and better styling. Curious to see your thoughts on this as well @Addison-Stavlo @jeyip

Part of the confusion here is the naming we're using.

```
'keep-active': keepActive && isCompleted
'is-completed': isCompleted && ! keepActive
```

We could probably rename things to help with the confusion. An idea for the former might be `completed-and-active`, while the latter might be `completed-and-inactive`, or they could be something else entirely if folks have better suggestions

> I'm not entirely sure what the solution is here. I think at root, we really still need to simplify the various states for each task

I agree with this thought. Can we create a follow-up task to step back and re-evaluate how to simplify the different task states?

Comment (jeyip) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx]: > Can we create a follow-up task to step back and re-evaluate how to simplify the different task states?

@Sharpirate Were you able to make a follow-up task for this?

**Comment by jeyip** [comment]
## Testing
### Requirements
- [x] Hovering with the mouse and focusing with the keyboard on all active links turns them blue
- [x] Tasks redirect to the proper pages when pressing the "enter" key

### Browsers
- [x] Edge
- [x] Firefox
- [x] Chrome

**Review by jeyip — APPROVED** [review]
